libraries/ms-drivers/test/test_adt7476a_fan_controller.c

```diff
@@ -0,0 +1,154 @@
+#include "adt7476a_fan_controller.h"
+#include "adt7476a_fan_controller_defs.h"
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_I2C_PORT I2C_PORT_2
+#define TEST_I2C_ADDRESS 0x74
+#define TEST_FAN_PWM_SPEED_1 0x03
+#define TEST_FAN_PWM_SPEED_2 0x04
+#define TEST_FAN_PWM_EXPECTED_SPEED 0x80
+
+#define TEST_CONFIG_PIN_I2C_SCL \
+  { GPIO_PORT_B, 10 }
+#define TEST_CONFIG_PIN_I2C_SDA \
+  { GPIO_PORT_B, 11 }
+
+typedef struct Adt7476aMockRegisters {
+  uint8_t PWM_CONFIG_1;
+  uint8_t PWM_CONFIG_2;
+  uint8_t SMBALERT_PIN;
+  uint8_t PWM_SPEED_1;
+  uint8_t PWM_SPEED_2;
+  uint8_t INTERRUPT_STATUS_1;
+  uint8_t INTERRUPT_STATUS_2;
+} Adt7476aMockRegisters;
+
+Adt7476aMockRegisters MockRegisters;
+Adt7476aStorage MockStorage;
+Adt7476aStorage storage;
```

Comment:
should be marked static and prefixed `s_` to match convention, should also use `snake_case` for variable names